account_move_line_analytic_group/hooks.py

```diff
@@ -0,0 +1,30 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openupgradelib import openupgrade
+import logging
+
+logger = logging.getLogger(__name__)
+logger.info(
+    "Creating columns: account_move_line."
+)
+
+def pre_init_hook(cr):
+    openupgrade.logged_query(
+        cr, "ALTER TABLE account_move_line ADD COLUMN IF NOT EXISTS analytic_account_group_id INT"
+    )
+    openupgrade.logged_query(
+        cr,
+        """
+            UPDATE account_move_line
+            SET analytic_account_group_id = COALESCE(subquery.group_id, NULL)
```

Comment:
You should also add root group initialization (good luck with PostgreSQL split, I don't know if exists)